### PR TITLE
Templating

### PR DESCRIPTION
**Pull request by lu-zero**
- [x] Convert the current template to tera
- [x] Add and option to pass a custom template
- [x] Refactor the template in a `base.tera` with good defaults and let the user rely on it
- [x] Document everything

### CODE REVIEW

File: src/base.tera [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{%- block header -%}
+# Copyright 2017-{{ this_year }} Gentoo Authors
```
Comment (gyakovlev) [inline src/base.tera:2]: just leave current year, it's ok for new ebuilds.

File: src/base.tera [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+# Auto-Generated by cargo-ebuild {{ cargo_ebuild_ver }}
+{% endblock %}
+EAPI={%- block eapi -%}7{%- endblock %}
```
Comment (gyakovlev) [inline src/base.tera:7]: let's use `8` by default.

File: src/base.tera [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+{%- endfor -%}"
+{%- endblock %}
+
+inherit {% block inherit -%}cargo{%- endblock %}
+
+DESCRIPTION={%- block description -%}"{{ description | trim }}"{%- endblock %}
+# Double check the homepage as the cargo_metadata crate
+# does not provide this value so instead repository is used
+HOMEPAGE={%- block homepage -%}"{{ homepage }}"{%- endblock %}
+SRC_URI={%- block src_uri -%}{% raw -%}"$(cargo_crate_uris ${CRATES})"{%- endraw %}{%- endblock %}
```
Comment (gyakovlev) [inline src/base.tera:22]: let's use  just `$(cargo_crate_uris)` as it automagically uses $CRATES nowadays, passing is optional.

File: src/base.tera [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# License set may be more restrictive as OR is not respected
+# use cargo-license for a more accurate license picture
+LICENSE={%- block license -%}"{{ license }}"{%- endblock %}
+SLOT={%- block slot -%}"0"{%- endblock %}
+KEYWORDS={%- block keyword -%}"~amd64"{%- endblock %}
+{% block variables -%}
+RESTRICT="mirror"
```
Comment (gyakovlev) [inline src/base.tera:29]: this restriction is ok and makes sense for overlays, but is discouraged for `::gentoo`
but I can edit the template later and add a simple comment that users should maybe remove it if ebuild is for main repo.

**Review by gyakovlev — CHANGES_REQUESTED** [review]
left some  template comments.
also, can you sign-off commits please?
it's required for gentoo namespace.
I've added a github check for it btw.

**Comment by lu-zero** [comment]
I rebased and updated the template according to your suggestion, let me know if it works for you.

**Pull request closed without merging**